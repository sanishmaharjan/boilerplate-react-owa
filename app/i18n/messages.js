export default {
    en: {
        LOGOUT: 'Logout',
        CHOOSE_LANGUAGE: 'Choose language',
        SUCCESS: 'Success',
        ERROR: 'Error',
        INFO: 'Info',
        HOME_PAGE: 'Home page',
        BLINK_PAGE: 'Blink page',
        USER_INFORMATION: 'User Information',
        USER_UUID: 'User Uuid',
        NAME: 'Name',
        USER_NAME: 'User Name',
        NOTIFICATION: 'Notification',
        MESSAGE: 'Message'
    },
    es: {
        LOGOUT: 'Cerrar sesión',
        CHOOSE_LANGUAGE: 'Elige lengua',
        SUCCESS: 'Éxito',
        ERROR: 'Error',
        INFO: 'Información',
        HOME_PAGE: 'Página de inicio',
        BLINK_PAGE: 'Página de parpadeo',
        USER_INFORMATION: 'informacion del usuario',
        USER_UUID: 'Usuario Uuid',
        NAME: 'Nombre',
        USER_NAME: 'Nombre de usuario',
        NOTIFICATION: 'Notificación',
        MESSAGE: 'Mensaje'
    },
    fr: {
        LOGOUT: 'Connectez - Out',
        CHOOSE_LANGUAGE: 'Choisissez la langue',
        SUCCESS: 'Succès',
        ERROR: 'Erreur',
        INFO: 'Info',
        HOME_PAGE: 'Page d\'accueil',
        BLINK_PAGE: 'Blink page',
        USER_INFORMATION: 'informations de l\'utilisateur',
        USER_UUID: 'Utilisateur Uuid',
        NAME: 'prénom',
        USER_NAME: 'Nom d\'utilisateur',
        NOTIFICATION: 'Notification',
        MESSAGE: 'Message'
    },
    pt: {
        LOGOUT: 'Sair',
        CHOOSE_LANGUAGE: 'Escolha o seu idioma',
        SUCCESS: 'Sucesso',
        ERROR: 'Erro',
        INFO: 'Info',
        HOME_PAGE: 'Pagina inicial',
        BLINK_PAGE: 'Página de piscar',
        USER_INFORMATION: 'Informação do usuário',
        USER_UUID: 'Usuário Uuid',
        NAME: 'Nome',
        USER_NAME: 'Nome de usuário',
        NOTIFICATION: 'Notificação',
        MESSAGE: 'mensagem'
    }
};
