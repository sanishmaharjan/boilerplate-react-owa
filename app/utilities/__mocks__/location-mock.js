Object.defineProperties(window.location, {
    host: {
        writable: true,
        value: '192.168.33.10'
    },
    hostname: {
        writable: true,
        value: '192.168.33.10'
    },
    href: {
        writable: true,
        value: 'https://192.168.33.10/openmrs/owa/boilerplate/index.html'
    },
    origin: {
        writable: true,
        value: 'https://192.168.33.10'
    },
    pathname: {
        writable: true,
        value: '/openmrs/owa/boilerplate/index.html'
    },
    port: {
        writable: true,
        value: ''
    },
    protocol: {
        writable: true,
        value: 'https:'
    }
});
