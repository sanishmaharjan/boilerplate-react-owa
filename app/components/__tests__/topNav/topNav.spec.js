import React from 'react';
import {mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import {IntlProvider} from 'react-intl';

import TopNav from 'components/topNav';
import messages from 'i18n/messages';

require('utilities/__mocks__/location-mock');
const intlProvider = new IntlProvider({locale: 'en', messages: messages['en']}, {});
const {intl} = intlProvider.getChildContext();
const testData = {
    context: {
        router: {
            history: {
                action: 'PUSH',
                push: jest.fn(),
                replace: jest.fn(),
                go: jest.fn(),
                goBack: jest.fn(),
                goForward: jest.fn(),
                createHref: jest.fn(),
                block: jest.fn(),
                length: 1,
                location: {
                    hash: '/blink',
                    key: 'd0o5f3',
                    pathname: 'https://192.168.33.10/openmrs/owa/boilerplate/index.html',
                    search: ''
                }
            },
            route: {
                location: {
                    hash: '/blink',
                    key: 'd0o5f3',
                    pathname: '/openmrs/owa/boilerplate/index.html',
                    search: ''
                },
                match: {
                    isExact: true,
                    params: {},
                    path: '/blink',
                    url: '/blink'
                }
            }
        },
        intl: intl
    }
};

describe('TopNav', () => {
    it('renders correctly', () => {
        const topNav = mount(<TopNav path="/blink" />, {context: testData.context});

        expect(topNav.find('ul').find('li').length).toBe(3);
        expect(
            topNav
                .find('a.active')
                .text()
                .trim()
        ).toBe('Blink page');
        expect(shallowToJson(topNav)).toMatchSnapshot();
    });
});
