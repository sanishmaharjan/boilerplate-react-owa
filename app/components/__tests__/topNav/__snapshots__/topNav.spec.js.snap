// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNav renders correctly 1`] = `
<TopNav
  path="/blink"
>
  <nav>
    <ul
      className="title-section"
    >
      <li>
        <a
          href="https://192.168.33.10/openmrs"
        >
          <i
            aria-hidden="true"
            className="fa fa-home"
          />
        </a>
      </li>
      <li>
        <Link
          className=""
          replace={false}
          to="/"
        >
          <a
            className=""
            onClick={[Function]}
          >
            Home page
          </a>
        </Link>
      </li>
      <li>
        <Link
          className="active"
          replace={true}
          to="/blink"
        >
          <a
            className="active"
            onClick={[Function]}
          >
            Blink page
          </a>
        </Link>
      </li>
    </ul>
  </nav>
</TopNav>
`;
